pkg/resources/istioingress/meshgateway.go

```diff
@@ -42,33 +43,82 @@ func (r *Reconciler) meshgateway(log logr.Logger, externalListenerConfig v1beta1
-						Resources:      ingressConfig.IstioIngressConfig.GetResources(),
-						NodeSelector:   ingressConfig.IstioIngressConfig.NodeSelector,
-						Tolerations:    ingressConfig.IstioIngressConfig.Tolerations,
-						PodAnnotations: ingressConfig.IstioIngressConfig.GetAnnotations(),
+		Spec: &istioOperatorApi.IstioMeshGatewaySpec{
+			Deployment: &istioOperatorApi.BaseKubernetesResourceConfig{
+				Metadata: &istioOperatorApi.K8SObjectMeta{
+					Labels:      labelsForIstioIngress(r.KafkaCluster.Name, eListenerLabelName),
+					Annotations: ingressConfig.IstioIngressConfig.GetAnnotations(),
+				},
+				//Image:                "",
+				Env: ingressConfig.IstioIngressConfig.Envs,
+				Resources: &istioOperatorApi.ResourceRequirements{
+					Limits: map[string]*istioOperatorApi.Quantity{
+						"cpu":    {Quantity: *ingressResources.Limits.Cpu()},
+						"memory": {Quantity: *ingressResources.Limits.Memory()},
+					},
+					Requests: map[string]*istioOperatorApi.Quantity{
+						"cpu":    {Quantity: *ingressResources.Requests.Cpu()},
+						"memory": {Quantity: *ingressResources.Requests.Memory()},
 					},
 				},
-				ServiceType:                  ingressConfig.GetServiceType(),
-				ServiceExternalTrafficPolicy: ingressConfig.ExternalTrafficPolicy,
+				NodeSelector: ingressConfig.IstioIngressConfig.NodeSelector,
+				//	Affinity:             &corev1.Affinity{},
+				SecurityContext: &corev1.SecurityContext{
+					RunAsUser:    util.Int64Pointer(0),
+					RunAsGroup:   util.Int64Pointer(0),
+					RunAsNonRoot: util.BoolPointer(false),
+				},
+				//	ImagePullPolicy:      "",
+				//	ImagePullSecrets:     []corev1.LocalObjectReference{},
+				//	PriorityClassName:    "",
+				Tolerations: ingressConfig.IstioIngressConfig.Tolerations,
+				//	Volumes:              []corev1.Volume{},
+				//	VolumeMounts:         []corev1.VolumeMount{},
+				Replicas: &istioOperatorApi.Replicas{
+					Count: util.Int32Pointer(ingressConfig.IstioIngressConfig.GetReplicas()),
+					Min:   util.Int32Pointer(ingressConfig.IstioIngressConfig.GetReplicas()),
+					Max:   util.Int32Pointer(ingressConfig.IstioIngressConfig.GetReplicas()),
+					//		TargetCPUUtilizationPercentage: new(int32),
+				},
+				// PodMetadata:          &istioOperatorApi.K8SObjectMeta{},
+				// PodDisruptionBudget:  &istioOperatorApi.PodDisruptionBudget{},
+				// DeploymentStrategy:   &istioOperatorApi.DeploymentStrategy{},
+				// PodSecurityContext:   &corev1.PodSecurityContext{},
+				// LivenessProbe:        &corev1.Probe{},
+				// ReadinessProbe:       &corev1.Probe{},
 			},
-			Ports: generateExternalPorts(r.KafkaCluster,
-				util.GetBrokerIdsFromStatusAndSpec(r.KafkaCluster.Status.BrokersState, r.KafkaCluster.Spec.Brokers, log),
-				externalListenerConfig, log, ingressConfigName, defaultIngressConfigName),
-			Type: istioOperatorApi.GatewayTypeIngress,
+			Service: &istioOperatorApi.Service{
+				Metadata: &istioOperatorApi.K8SObjectMeta{},
```

Comment:
I think the ` ingressConfig.GetServiceAnnotations()` must be passed onto the service.